splink/linker.py

```diff
@@ -469,6 +463,27 @@ def _execute_sql_against_backend(self, sql, templated_name, physical_name):
             f"_execute_sql_against_backend not implemented for {type(self)}"
         )
 
+    def _do_execute_sql(self, final_sql, templated_name, physical_name):
```

Comment:
I'd probably rename this to be `_execute_sql()`.

Reply:
I called it this because I don't want users to directly call this method. They should only be calling `_execute_sql_against_backend`, and then the internals of the class go and delegate to this method.

What about:

1. rename _execute_sql_against_backend -> _execute_sql(): This is what users of the class should actually be using, so make it short and obvious.
2. Keep _do_execute_sql and _log_and_do_execute_sql named as they are.

I would like to somehow hide _do_execute_sql and _log_and_do_execute_sql better so they look "More" internal than just a single leading _, but I'm not sure how to do that. I think some magix with metaclasses would work, but I think we do NOT want to go down that route. Maybe a double leading underscore? But I don't really understand how that works, some more magic name mangling happens.